src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/output_format_converter.py

```diff
@@ -0,0 +1,275 @@
+            payload = {"prompt": text_content}
+
+            if add_model_name:
+                payload["model"] = model
+            if add_stream:
+                payload["stream"] = True
+            if output_tokens_mean != -1:
+                payload["max_tokens"] = int(
+                    random.gauss(output_tokens_mean, output_tokens_stddev)
+                )
+            payload.update(extra_inputs)
+
+            pa_json["data"].append({"payload": [payload]})
+
+        return pa_json
+
+
+class OpenAIEmbeddingsConverter(BaseConverter):
+    def convert(
+        self,
+        generic_dataset: Dict,
+        add_model_name: bool,
+        add_stream: bool,
+        extra_inputs: Dict,
+        output_tokens_mean: int,
+        output_tokens_stddev: int,
+        output_tokens_deterministic: bool,
+        model_name: list,
+        model_selection_strategy: ModelSelectionStrategy,
+    ) -> Dict:
+        pa_json: Dict = {"data": []}
+
+        for index, row in enumerate(generic_dataset["rows"]):
+            text_content = row["row"]["text_input"]
+            model = self._select_model_name(model_name, index, model_selection_strategy)
+            payload: Dict = {"input": [text_content], "model": model}
+
+            if add_stream:
+                payload["stream"] = True
+            payload.update(extra_inputs)
+
+            pa_json["data"].append({"payload": [payload]})
+
+        return pa_json
+
+
+class RankingsConverter(BaseConverter):
+    def convert(
+        self,
+        generic_dataset: Dict,
+        add_model_name: bool,
+        add_stream: bool,
+        extra_inputs: Dict,
+        output_tokens_mean: int,
+        output_tokens_stddev: int,
+        output_tokens_deterministic: bool,
+        model_name: list,
+        model_selection_strategy: ModelSelectionStrategy,
+    ) -> Dict:
+        pa_json: Dict = {"data": []}
+
+        for index, row in enumerate(generic_dataset["rows"]):
+            if "query" not in row or "passages" not in row:
+                raise GenAIPerfException(
+                    "Expected keys 'query' and 'passages' not found in dataset row."
+                )
+
+            model = self._select_model_name(model_name, index, model_selection_strategy)
+            payload = {
+                "query": row["query"]["text_input"],
+                "passages": [p["text_input"] for p in row["passages"]],
+                "model": model,
+            }
+
+            if add_stream:
+                payload["stream"] = True
+            payload.update(extra_inputs)
+
+            pa_json["data"].append({"payload": [payload]})
+
+        return pa_json
+
+
+class VLLMConverter(BaseConverter):
+    def convert(
+        self,
+        generic_dataset: Dict,
+        add_model_name: bool,
+        add_stream: bool,
+        extra_inputs: Dict,
+        output_tokens_mean: int,
+        output_tokens_stddev: int,
+        output_tokens_deterministic: bool,
```

Comment:
do we need to specify output tokens using these arguments?

Reply:
They're going to be added in a future PR on this feature branch. I haven't implemented output token distributions yet, that's in future work.